app/uk/gov/hmrc/cataloguefrontend/repository/RepositoriesController.scala

```diff
@@ -43,64 +43,75 @@ class RepositoriesController @Inject() (
 ) extends FrontendController(mcc)
   with CatalogueAuthBuilders {
 
-  def allRepositories(name: Option[String], team: Option[String], repoType: Option[String]): Action[AnyContent] =
+  def allRepositories(name: Option[String] = None, team: Option[String] = None, archived: Option[Boolean] = None, repoTypeFilter: Option[String] = None): Action[AnyContent] =
     BasicAuthAction.async { implicit request =>
-      import SearchFiltering._
-
       val allTeams =
         teamsAndRepositoriesConnector
           .allTeams
           .map(_.sortBy(_.name.asString))
 
+      val (repoType, serviceType) = buildQueryParam(repoTypeFilter)
+
       val allRepositories =
         teamsAndRepositoriesConnector
-          .allRepositories
+          .allRepositories(name.filterNot(_.isEmpty), team.filterNot(_.isEmpty), archived, repoType, serviceType)
```

Comment:
empty strings sounds like an upstream bug - I think we should filter them out when entering the system rather than here

Reply:
This is probably fine, I missed we're in the Controller